src/client/cli/cmd/delete.cpp

```diff
@@ -104,17 +105,44 @@ mp::ParseCode cmd::Delete::parse_args(mp::ArgParser* parser)
         return parse_code;
 
     request.set_purge(parser->isSet(purge_option));
+
+    bool instance_found = false, snapshot_found = false;
     for (const auto& item : add_instance_and_snapshot_names(parser))
     {
-        if (item.has_snapshot_name() && !request.purge())
-        {
-            cerr << "Snapshots can only be purged (after deletion, they cannot be recovered). Please use the `--purge` "
-                    "flag if that is what you want.\n";
-            return mp::ParseCode::CommandLineError;
-        }
+        instance_found = instance_found || !item.has_snapshot_name();
+        snapshot_found = snapshot_found || item.has_snapshot_name();
 
         request.add_instances_snapshots()->CopyFrom(item);
     }
 
+    if (!request.purge() && snapshot_found)
+    {
+        cerr << "Snapshots can only be purged (after deletion, they cannot be recovered). Please use the `--purge` "
+                "flag if that is what you want.\n";
+        return mp::ParseCode::CommandLineError;
+    }
+    else if (request.purge() && instance_found && snapshot_found)
+    {
+        if (term->is_live())
+        {
+            try
+            {
+                if (!confirm_action(term, "warning: Specifying both instances and snapshots together will also "
+                                          "permanently delete instances.\n"
+                                          "Are you sure you want to continue? (Yes/no)"))
+                    return mp::ParseCode::CommandFail;
+            }
+            catch (const mp::PromptException& e)
+            {
+                std::cerr << e.what() << std::endl;
+                return ParseCode::CommandLineError;
+            }
```

Comment:
Why did you opt for a confirmation question on `--purge`? I see a few problems with this:
  - it would only be shown in interactive terminals, but succeed silently in scripts
  - it would warn when the user added `--purge` to begin with, on their own
  - it is another interactive question (I am a little wary of proliferating those)
  - the confirmation is asked only in a very specific case, making the usage asymmetric for no obvious reason (why ask for confirmation only if snapshots and instances are involved)?
  
IMO, the goal is only to avoid misguiding the user to purge instances inadvertently. If they choose to add the purge flag on their own, I don't think it makes a lot of sense to ask for confirmation.